zetaclient/chains/evm/observer/observer.go

```diff
@@ -332,16 +332,24 @@ func (ob *Observer) PostGasPrice(ctx context.Context) error {
 		ob.Logger().GasPrice.Err(err).Msg("Err SuggestGasPrice:")
 		return err
 	}
+
+	priorityFee, err := ob.evmClient.SuggestGasTipCap(ctx)
+	if err != nil {
+		ob.Logger().GasPrice.Err(err).Msg("unable to load priority fee")
+		// we want fallback to zero if we can't get the priority fee
```

Comment:
The `err` can be any one of RPC failures. To tell if a network supports EIP-1559 or not, we need to fetch block header and check if the `baseFee` field exists. You can refer to [Chainsafe's implementation](https://github.com/ChainSafe/chainbridge-core/blob/main/chains/evm/calls/evmgaspricer/london.go#L24) .

The logic would look like

1. Fetch header and check if the baseFee exists.
2. If it exists, then we MUST succeed in the `SuggestGasTipCap` query.
3. If it doesn't exist, the network doesn't support EIP-1559 and we fall back to zero.

Reply:
https://github.com/zeta-chain/node/pull/2483/commits/aa6de92d7d8dae05c29c029453f2b64228b34d0f